l10n_jp_partner_title_qweb/migrations/19.0.1.0.0/post-migration.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+
+def migrate(cr, version):
+    # Check that old columns still exist
+    cr.execute("""
+        SELECT column_name
+        FROM information_schema.columns
+        WHERE table_name='res_partner' AND column_name='title'
+    """)
+    if not cr.fetchone():
+        return
+    cr.execute("""
+        UPDATE res_partner p
+        SET honorific_title = COALESCE(t.shortcut, t.name),
+            honorific_title_position = t.display_position
+        FROM res_partner_title t
+        WHERE p.title = t.id
+    """)
```

Comment:
I actually doubt that res_partner_title table will survive through the core migration, but anyway.

```suggestion
    if not version:
        return
    # Only run if BOTH the old column and table still exist after the core migration.
    has_partner_title_col = openupgrade.column_exists(cr, "res_partner", "title")
    has_title_table = openupgrade.table_exists(cr, "res_partner_title")
    if not (has_partner_title_col and has_title_table):
        return
    openupgrade.logged_query(
        cr,
        """
        UPDATE res_partner AS p
           SET honorific_title = t.name,
               honorific_title_position = COALESCE(t.display_position, 'after')
          FROM res_partner_title AS t
         WHERE p.title = t.id
        """
    )
```

Reply:
I used Odoo’s standard functions column_exists and table_exists instead of the methods from openupgradelib to avoid adding an external library unless it’s absolutely necessary.
Please let me know if there’s any specific reason to use openupgradelib.